payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherViewModel.kt

```diff
@@ -59,13 +59,39 @@ internal class GooglePayLauncherViewModel(
     }
 
     @VisibleForTesting
-    suspend fun createPaymentDataRequest(): JSONObject {
-        val transactionInfo = createTransactionInfo(
-            stripeRepository.retrieveStripeIntent(
-                args.clientSecret,
-                requestOptions,
-            )
-        )
+    suspend fun createPaymentDataRequest(
+        args: GooglePayLauncherContract.Args
+    ): JSONObject {
+        val transactionInfo = when (args) {
+            is GooglePayLauncherContract.PaymentIntentArgs -> {
+                val paymentIntent = requireNotNull(
+                    stripeRepository.retrievePaymentIntent(
+                        args.clientSecret,
+                        requestOptions,
+                    )
+                ) {
```

Comment:
I think in paymentSHeet we return an Failure to the user if we can't find the payment intent

Reply:
The exception will be handled and communicated as a `Failure`